jammsen/docker-palworld-dedicated-server
# [Feat] Add rcon broadcast with spaces feature

# Add RCON broadcast command with spaces

## Summary

This pull request introduces changes to the RCON broadcast commands, using a new custom RCON binary.

The source code for the binary is available on my [Palworld Dedicated Server on Docker](https://github.com/thejcpalma/palworld-dedicated-server-docker) repository (pinned to version [`0.2.1`](https://github.com/thejcpalma/palworld-dedicated-server-docker/tree/0.2.1/src/custom_rcon_broadcast)).

Notable modifications include the removal of dashes from all broadcast messages and improvements to error handling when players with special characters join the server.

## Motivation and Context

The main goal was to provide the broadcast feature/command of Palworld's RCON, but having the ability to use spaces, so messages are clearer to read and pleasant overall. Both approaches of using dashes `-` or underscores `_` seemed ugly from the start, but was the initial approach when the problem was encountered.

## Description

- Updated RCON broadcast commands to use a custom RCON binary from my [repository](https://github.com/thejcpalma/palworld-dedicated-server-docker).
- Modified the Dockerfile build stage to include the compilation of the custom binary within the Golang image.
- Removed dashes from all broadcast messages to improve readability and consistency.
- When using the RCON `showplayers` command, stderr is now redirected to /dev/null to prevent log errors caused by players with special characters in their names.
- Updated `rconcli.sh` to accommodate the new custom binary for handling broadcast messages.
- Changed the output of `rconcli.sh` to use regular `echo` to avoid issues with ANSI codes from special characters causing misinterpretations in the player detection system.
- Conducted spell check on the `default.env` file for accuracy and consistency.

## Testing Instructions

1. Clone the repository.
2. Build the Docker image using the modified Dockerfile and new binary.
3. Join the server.
4. Execute various RCON broadcast commands using the updated `rconcli.sh`.
5. Test the RCON `showplayers` command with players (both with and without special characters in their names).
6. Verify that broadcast messages no longer contain dashes and are displayed correctly inside the server.
7. Ensure that the player detection system interprets the output from `rconcli` accurately.
8. Verify if all other dependent features work correctly (auto-update, auto-restart, ...)

## Checklist
